Ghidra/Features/Base/src/main/java/ghidra/program/flatapi/FlatProgramAPI.java

```diff
@@ -1514,6 +1514,58 @@ public final List<Symbol> getSymbols(String name, Namespace namespace) {
 	public final Namespace getNamespace(Namespace parent, String namespaceName) {
 		return currentProgram.getSymbolTable().getNamespace(namespaceName, parent);
 	}
+	
+	/**
+	 * Creates a new namespace with the given name contained inside the
+	 * specified parent namespace.
+	 * Pass <code>null</code> for namespace to indicate the global namespace.
+	 * If a Namespace or GhidraClass with the given name already exists, the
+	 * existing one will be returned.
+	 * @param parent the parent namespace, or null for global namespace
+	 * @param namespaceName the requested namespace's name
+	 * @return the namespace with the given name
+	 * @throws DuplicateNameException thrown if a {@link Library} symbol exists with the given name
+	 * @throws InvalidInputException if the name is invalid
+	 */
+	public final Namespace createNamespace(Namespace parent, String namespaceName)
+			throws DuplicateNameException, InvalidInputException {
+		SymbolTable symbolTable = currentProgram.getSymbolTable();
+		Namespace ns = symbolTable.getNamespace(namespaceName, parent);
+		if (ns != null) {
+			SymbolType type = ns.getSymbol().getSymbolType();
+			if (type == SymbolType.NAMESPACE || type == SymbolType.CLASS) {
+				return ns;
+			}
+		}
+		return symbolTable.createNameSpace(parent, namespaceName, SourceType.USER_DEFINED);
+	}
+	
+	/**
+	 * Creates a new GhidraClass with the given name contained inside the
+	 * specified parent namespace.
+	 * Pass <code>null</code> for namespace to indicate the global namespace.
+	 * If a GhidraClass with the given name already exists, the existing one will be returned
+	 * @param parent the parent namespace, or null for global namespace
+	 * @param className the requested classes name
+	 * @return the GhidraClass with the given name
+	 * @throws InvalidInputException if the name is invalid
+	 * @throws DuplicateNameException thrown if a {@link Library} or {@link Namespace}
+	 * symbol already exists with the given name.
+	 * @see SymbolTable#convertNamespaceToClass(Namespace) for converting an existing Namespace to
```

Comment:
May also want to link to `Namespace.getSymbol()` may be used to obtain the symbol to be converted.

Reply:
What would the symbol be used for?